### ISSUE TITLE
Different result about 3dNormals estimation when after performing affine transformation

### ISSUE THREAD

**Issue by wenxingguo** [PointCloudLibrary/pcl#5024]
**Describe the bug**
Different result about 3dNormals estimation when after performing affine transformation.

**Screenshots/Code snippets**
```c++
#include <iostream>
#include <pcl/io/pcd_io.h>
#include <pcl/common/transforms.h>
#include <pcl/common/angles.h>
#include <pcl/filters/voxel_grid.h>
#include <pcl/features/normal_3d_omp.h>
#include <pcl/kdtree/kdtree_flann.h>

Eigen::Affine3d affine(const Eigen::Vector3d& deg, const Eigen::Vector3d& offset){
		Eigen::Affine3d transform = Eigen::Affine3d::Identity();
		transform.rotate(Eigen::AngleAxisd(pcl::deg2rad(deg[0]), Eigen::Vector3d::UnitZ()));
		transform.rotate(Eigen::AngleAxisd(pcl::deg2rad(deg[1]), Eigen::Vector3d::UnitY()));
		transform.rotate(Eigen::AngleAxisd(pcl::deg2rad(deg[2]), Eigen::Vector3d::UnitX()));

		transform.translation() = offset;
		return  transform;
}

void computePointNormal(pcl::PointCloud<pcl::PointXYZ>::ConstPtr cloud, pcl::PointCloud<pcl::Normal>::Ptr cloud_normal){
		pcl::NormalEstimationOMP<pcl::PointXYZ, pcl::Normal> n;
		pcl::search::KdTree<pcl::PointXYZ>::Ptr kdtree_flann(new pcl::search::KdTree<pcl::PointXYZ>);
		kdtree_flann->setInputCloud(cloud);
		n.setInputCloud(cloud);
		n.setNumberOfThreads(8);
		n.setKSearch(250);
		n.setSearchMethod(kdtree_flann);
		n.compute(*cloud_normal);
}

int main (int argc, char** argv)
{
		pcl::PointCloud<pcl::PointXYZ>::Ptr cloud1(new pcl::PointCloud<pcl::PointXYZ>);
		pcl::PointCloud<pcl::PointXYZ>::Ptr cloud2(new pcl::PointCloud<pcl::PointXYZ>);
		pcl::io::loadPCDFile("./D1_save.pcd", *cloud1);
		
		//VoxelGrid
		pcl::VoxelGrid<pcl::PointXYZ> sor;
		sor.setLeafSize(0.0005, 0.0005, 0.0005);
		sor.setInputCloud(cloud1);
		sor.filter(*cloud1);

		//Affine transform
		Eigen::Affine3d transform = affine({-107.97, 6.82, 178.52}, {1396.93/1000.0, -1521.94/1000.0, 1535.11/1000.0});
		pcl::transformPointCloud(*cloud1, *cloud2, transform);

		//Estimate normals
		pcl::PointCloud<pcl::Normal>::Ptr cloud_normal1(new pcl::PointCloud<pcl::Normal>);
		pcl::PointCloud<pcl::Normal>::Ptr cloud_normal2(new pcl::PointCloud<pcl::Normal>);

		computePointNormal(cloud1, cloud_normal1);
		computePointNormal(cloud2, cloud_normal2);

		//savePCDFile
		pcl::PointCloud<pcl::PointXYZINormal>::Ptr tmp(new pcl::PointCloud<pcl::PointXYZINormal>);
		for(size_t i = 0; i < cloud1->size(); ++i){
				tmp->push_back(pcl::PointXYZINormal(
										cloud1->at(i).x, cloud1->at(i).y, cloud1->at(i).z,
										1,
										cloud_normal1->at(i).normal_x, cloud_normal1->at(i).normal_y, cloud_normal1->at(i).normal_z, cloud_normal1->at(i).curvature
										));
		}
		pcl::io::savePCDFile("cloud1_with_normal.pcd", *tmp);

		tmp->clear();
		for(size_t i = 0; i < cloud2->size(); ++i){
				tmp->push_back(pcl::PointXYZINormal(
										cloud2->at(i).x, cloud2->at(i).y, cloud2->at(i).z,
										1,
										cloud_normal2->at(i).normal_x, cloud_normal2->at(i).normal_y, cloud_normal2->at(i).normal_z, cloud_normal2->at(i).curvature
										));
		}
		pcl::io::savePCDFile("cloud2_with_normal.pcd", *tmp);
                return 0;
}
```
Get the result (pcl_viewer -normals 1 PCDFile.pcd)
![image](https://raw.githubusercontent.com/wenxingguo/pcl_img/main/pic.png)

PCD file
[file](https://github.com/wenxingguo/pcl_img/blob/main/D1_save.pcd)

**Your Environment (please complete the following information):**

 - OS: [macos big sur 11.6]
 - Compiler: [clang13]
 - PCL Version [1.12]

**Comment by wenxingguo** [comment]
use pcl_viewer -normals 1

**Comment by mvieth** [comment]
I would say this is either related to issue #4965 (for details see there, a pull request that fixes the problem is pending), or related to the behaviour that the normals are oriented towards the viewpoint (see e.g. [`setViewPoint`](https://pointclouds.org/documentation/classpcl_1_1_normal_estimation.html#a215cb3a84f03a4346556219121fce8e5))

**Comment by mvieth** [comment]
Ok, so I would say that this is not because of the oriented-towards-viewpoint behaviour, because then all normals would be affected similarly, and they would be just inverted (multiplied with -1). However in the image, only few normals are affected, and they are oriented more or less randomly.
So this is very likely the same problem as in the other issue I linked above. I tested with the fix in pull request #4983 , then the normals look correct even after the transformation.
I am closing this issue since the problem is already documented in the other issue, and a pull request is on its way
